aws/resource_aws_ssm_patch_baseline.go

```diff
@@ -181,6 +196,14 @@ func resourceAwsSsmPatchBaselineCreate(d *schema.ResourceData, meta interface{})
 		params.ApprovalRules = expandAwsSsmPatchRuleGroup(d)
 	}
 
+	if v, ok := d.GetOk("approved_patches_enable_non_security"); ok && aws.StringValue(params.OperatingSystem) != ssm.OperatingSystemWindows {
```

Comment:
Perhaps we could error or something if `approved_patches_enable_non_security` == true and `operating_system` == WINDOWS instead of silently continuing? Otherwise the terraform code won't match what is being sent in the request to AWS.

Reply:
I can remove the os condition, the request will fail if OS is windows on AWS side.

Reply:
I removed the check, test is passing with linux os.